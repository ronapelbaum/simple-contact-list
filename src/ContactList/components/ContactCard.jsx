import React, { useState } from 'react';
import { Card, CardContent, CardMedia, Typography } from '@material-ui/core';
import { Rating } from '@material-ui/lab';
import Avatar from './Avatar';
import { Contact } from '../prop-types';
import styles from './ContactCard.module.scss';

const DEFAULT_IMAGE = 'http://caillouetland.com/wp-content/uploads/2017/07/avatar-blank.png';
const Field = ({ label, val }) => val ? 
  <div>
    <span className={styles.label}>{label}:</span>
    {val}
  </div>
  :
  null
;

const ContactCard = ({ contact }) => {
  const [showMore, setShowMore] = useState(false);

  return (
    <Card
      className={styles.card}
      onMouseEnter={() => setShowMore(true)}
      onMouseLeave={() => setShowMore(false)}
    >
      <CardContent className={styles.content}>
        <div className={styles.image}>
          <CardMedia component="img" image={contact.profile_image || DEFAULT_IMAGE} />
        </div>
        <div>
          <Avatar
            type={contact.driverType}
            className={styles.avatar}
          />
          <Typography variant="h5" component="h2">
            {contact.name}
          </Typography>
          <Rating
            value={Number(contact.driverRank)}
            size="small"
            readOnly
          />
          { showMore &&
            <React.Fragment>
              <Field label="phone number" val={contact.phone} />
              <Field label="email" val={contact.email} />
            </React.Fragment>
          }
        </div>
        
      </CardContent>
    </Card>
  );
};

ContactCard.propTypes = {
  contact: Contact.isRequired,
};

export default ContactCard;
