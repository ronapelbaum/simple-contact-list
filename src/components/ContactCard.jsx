import React from 'react';
import PropTypes from 'prop-types';
import * as Actions from '../actions';
import { Contact } from '../proptypes/Contact';

const ContactCard = ({ name, phone }) => (
  <div>
    <div>{name}</div>
    <div>{phone}</div>
  </div>
);

ContactCard.propTypes = Contact;

export default ContactCard;
