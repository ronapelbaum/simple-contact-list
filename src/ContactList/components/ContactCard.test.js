import React from "react";
import { shallow, mount } from "enzyme";
import ContactCard from "./ContactCard";

describe("ContactCard.jsx suite", () => {
  const contactList = [
    {
      name: "Luke Peterson",
      driverType: "Professional",
      driverRank: "4",
      phone: "123456     7890",
      email: "driver.luke@gmail.com",
      profile_image:
        "http://images.unsplash.com/photo-1504575070132-986227df99f7?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=7ecf0d8a55890fdb50eba8d4def9645d",
    },
    {
      name: "Ziv Avni",
      driverType: "Citizen",
      driverRank: "5",
      phone: "123.456.7895",
      email: "driver.ziv@gmail.com",
    },
    {
      name: "Mustafa Hussain",
      driverType: "Citizen",
      driverRank: "5",
      phone: "123.456.7900",
      email: "driver.mustafa@gmail.com",
      profile_image: "",
    },
  ];
  it("renders correctly for contact 1", () => {
    expect(shallow(<ContactCard contact={contactList[0]} />)).toMatchSnapshot();
  });
  it("renders correctly for contact 2", () => {
    expect(shallow(<ContactCard contact={contactList[1]} />)).toMatchSnapshot();
  });
  it("renders correctly for contact 3", () => {
    expect(shallow(<ContactCard contact={contactList[2]} />)).toMatchSnapshot();
  });
  // it('renders correctly with mouseEnter', () => {
  //   const wrapper = mount(<ContactCard contact={contactList[0]} />);
  //   console.log('rapelbaum - ContactCard.test', wrapper);
  //   wrapper.find('.MuiCard-root').simulate('mouseEnter');
    
  //   // expect(wrapper).toMatchSnapshot();
  // });
});
