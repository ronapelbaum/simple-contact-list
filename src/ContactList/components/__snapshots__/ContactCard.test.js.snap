// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard.jsx suite renders correctly for contact 1 1`] = `
"<WithStyles(ForwardRef(Card)) className=\\"card\\" onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
  <WithStyles(ForwardRef(CardContent)) className=\\"content\\">
    <div className=\\"image\\">
      <WithStyles(ForwardRef(CardMedia)) component=\\"img\\" image=\\"http://images.unsplash.com/photo-1504575070132-986227df99f7?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=7ecf0d8a55890fdb50eba8d4def9645d\\" />
    </div>
    <div>
      <Avatar type=\\"Professional\\" className=\\"avatar\\" />
      <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\">
        Luke Peterson
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Rating)) value={4} size=\\"small\\" readOnly={true} />
    </div>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>"
`;

exports[`ContactCard.jsx suite renders correctly for contact 2 1`] = `
"<WithStyles(ForwardRef(Card)) className=\\"card\\" onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
  <WithStyles(ForwardRef(CardContent)) className=\\"content\\">
    <div className=\\"image\\">
      <WithStyles(ForwardRef(CardMedia)) component=\\"img\\" image=\\"http://caillouetland.com/wp-content/uploads/2017/07/avatar-blank.png\\" />
    </div>
    <div>
      <Avatar type=\\"Citizen\\" className=\\"avatar\\" />
      <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\">
        Ziv Avni
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Rating)) value={5} size=\\"small\\" readOnly={true} />
    </div>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>"
`;

exports[`ContactCard.jsx suite renders correctly for contact 3 1`] = `
"<WithStyles(ForwardRef(Card)) className=\\"card\\" onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
  <WithStyles(ForwardRef(CardContent)) className=\\"content\\">
    <div className=\\"image\\">
      <WithStyles(ForwardRef(CardMedia)) component=\\"img\\" image=\\"http://caillouetland.com/wp-content/uploads/2017/07/avatar-blank.png\\" />
    </div>
    <div>
      <Avatar type=\\"Citizen\\" className=\\"avatar\\" />
      <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\">
        Mustafa Hussain
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Rating)) value={5} size=\\"small\\" readOnly={true} />
    </div>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>"
`;

exports[`ContactCard.jsx suite renders correctly with mouseEnter 1`] = `
"<ContactCard contact={{...}}>
  <WithStyles(ForwardRef(Card)) className=\\"card\\" onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
    <ForwardRef(Card) classes={{...}} className=\\"card\\" onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
      <WithStyles(ForwardRef(Paper)) className=\\"MuiCard-root card\\" elevation={1} onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
        <ForwardRef(Paper) classes={{...}} className=\\"MuiCard-root card\\" elevation={1} onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
          <div className=\\"MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded\\" onMouseEnter={[Function: onMouseEnter]} onMouseLeave={[Function: onMouseLeave]}>
            <WithStyles(ForwardRef(CardContent)) className=\\"content\\">
              <ForwardRef(CardContent) classes={{...}} className=\\"content\\">
                <div className=\\"MuiCardContent-root content\\">
                  <div className=\\"image\\">
                    <WithStyles(ForwardRef(CardMedia)) component=\\"img\\" image=\\"http://images.unsplash.com/photo-1504575070132-986227df99f7?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=7ecf0d8a55890fdb50eba8d4def9645d\\">
                      <ForwardRef(CardMedia) classes={{...}} component=\\"img\\" image=\\"http://images.unsplash.com/photo-1504575070132-986227df99f7?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=7ecf0d8a55890fdb50eba8d4def9645d\\">
                        <img className=\\"MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img\\" style={[undefined]} src=\\"http://images.unsplash.com/photo-1504575070132-986227df99f7?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=7ecf0d8a55890fdb50eba8d4def9645d\\" />
                      </ForwardRef(CardMedia)>
                    </WithStyles(ForwardRef(CardMedia))>
                  </div>
                  <div>
                    <Avatar type=\\"Professional\\" className=\\"avatar\\">
                      <ForwardRef(SvgProfessional) className=\\"avatar\\">
                        <svg className=\\"avatar\\">
                          professional.svg
                        </svg>
                      </ForwardRef(SvgProfessional)>
                    </Avatar>
                    <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\">
                      <ForwardRef(Typography) classes={{...}} variant=\\"h5\\" component=\\"h2\\">
                        <h2 className=\\"MuiTypography-root MuiTypography-h5\\">
                          Luke Peterson
                        </h2>
                      </ForwardRef(Typography)>
                    </WithStyles(ForwardRef(Typography))>
                    <WithStyles(ForwardRef(Rating)) value={4} size=\\"small\\" readOnly={true}>
                      <ForwardRef(Rating) classes={{...}} value={4} size=\\"small\\" readOnly={true}>
                        <span onMouseMove={[Function: handleMouseMove]} onMouseLeave={[Function: handleMouseLeave]} className=\\"MuiRating-root MuiRating-sizeSmall MuiRating-readOnly\\" role=\\"img\\" aria-label=\\"4 Stars\\">
                          <span>
                            <IconContainer value={1} className=\\"MuiRating-icon MuiRating-iconFilled\\">
                              <span className=\\"MuiRating-icon MuiRating-iconFilled\\">
                                <ForwardRef fontSize=\\"inherit\\">
                                  <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                    <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                      <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role={[undefined]}>
                                        <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                            </IconContainer>
                          </span>
                          <span>
                            <IconContainer value={2} className=\\"MuiRating-icon MuiRating-iconFilled\\">
                              <span className=\\"MuiRating-icon MuiRating-iconFilled\\">
                                <ForwardRef fontSize=\\"inherit\\">
                                  <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                    <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                      <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role={[undefined]}>
                                        <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                            </IconContainer>
                          </span>
                          <span>
                            <IconContainer value={3} className=\\"MuiRating-icon MuiRating-iconFilled\\">
                              <span className=\\"MuiRating-icon MuiRating-iconFilled\\">
                                <ForwardRef fontSize=\\"inherit\\">
                                  <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                    <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                      <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role={[undefined]}>
                                        <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                            </IconContainer>
                          </span>
                          <span>
                            <IconContainer value={4} className=\\"MuiRating-icon MuiRating-iconFilled\\">
                              <span className=\\"MuiRating-icon MuiRating-iconFilled\\">
                                <ForwardRef fontSize=\\"inherit\\">
                                  <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                    <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                      <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role={[undefined]}>
                                        <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                            </IconContainer>
                          </span>
                          <span>
                            <IconContainer value={5} className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                              <span className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                <ForwardRef fontSize=\\"inherit\\">
                                  <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                    <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                      <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role={[undefined]}>
                                        <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                            </IconContainer>
                          </span>
                        </span>
                      </ForwardRef(Rating)>
                    </WithStyles(ForwardRef(Rating))>
                    <Field label=\\"phone number\\" val=\\"123456     7890\\">
                      <div>
                        <span className=\\"label\\">
                          phone number
                          :
                        </span>
                        <span title=\\"123456     7890\\">
                          123456     7890
                        </span>
                      </div>
                    </Field>
                    <Field label=\\"email\\" val=\\"driver.luke@gmail.com\\">
                      <div>
                        <span className=\\"label\\">
                          email
                          :
                        </span>
                        <span title=\\"driver.luke@gmail.com\\">
                          driver.luke@gmail.com
                        </span>
                      </div>
                    </Field>
                  </div>
                </div>
              </ForwardRef(CardContent)>
            </WithStyles(ForwardRef(CardContent))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</ContactCard>"
`;
